lib/3scale/backend/application.rb

```diff
@@ -230,11 +230,12 @@ def load_all_usage_limits
       # Loads the usage limits affected by the metrics received, that is, the
       # limits that are defined for those metrics plus all their ancestors in
       # the metrics hierarchy.
+      # Raises MetricInvalid when a metric does not exist.
       def load_usage_limits_affected_by(metric_names)
         metric_ids = metric_names.flat_map do |name|
           [name] + Metric.ascendants(service_id, name)
         end.uniq.map do |name|
-          Metric.load_id(service_id, name)
+          Metric.load_id(service_id, name) || raise(MetricInvalid.new(name))
```

Comment:
I think this is problematic - any time the above fails it will assume the issue is an invalid metric. Is the method returning falsey always going to be the case of invalid metrics? Are we using this method in any other context where embedding the raised exception in `Metric.load_id` for falsey values would not work?

Reply:
`Metric.load_id` can return:
1) An ID when the metric exists
2) Nil when the metric does no exist.
3) A Redis Exception.
So yes, we can assume that if it's nil, it's because the metric does no exist.

Reply:
Sure - the last question is still unanswered. Does it make sense to raise within `Metric.load_id`?

Reply:
I remember that we had this discussion in the past and the answer was no, but now there's only one caller apart from this one and it also raises: https://github.com/3scale/apisonator/blob/e965a2228c331c21b159925d16888a5f473ef26c/lib/3scale/backend/metric/collection.rb#L78

So it looks like it should work, but it needs some testing to be sure. Let's address that in a separate PR.

Reply:
Sounds good! :+1: